File: design-patterns/ts/factory/transport/plane.ts
// plane.ts

import Transport from "./transport";

class Plane implements Transport {
    start(): void {
        console.log("Plane started");
    }

    stop(): void {
        console.log("Plane Stopped");
    }

    repair(): void {
        console.log("Plane Repair");
    }
}

export default Plane;

File: design-patterns/go/factory/transport/plane.go
// plane.go

package main

import "fmt"

type Plane struct {
}

func NewPlane() (plane *Plane) {
	plane = &Plane{}
	return
}

func (plane *Plane) Repair() {
	fmt.Println("Plane Repair")
}

func (plane *Plane) Start() {
	fmt.Println("Plane started")
}

func (plane *Plane) Stop() {
	fmt.Println("Plane Stopped")
}